### DISCUSSION TITLE
Add dedicated field name option to circumvent Python keyword conflicts

### DISCUSSION THREAD

**Discussion by inkychris** [pydantic/pydantic#2777] [Feature Request]
This topic is discussed in #153, however; the solutions proposed in that thread don't *elegantly* solve the problem in all cases. Consider the following example where a model has a field name that conflicts with a Python keyword, and another field that is badly named (and a user has no control over the model field names themselves):
```python
class Model(pydantic.BaseModel):
    if_: str = pydantic.Field(None, alias='condition')
    badly_named_field: str = pydantic.Field(None, alias='better_name')
```
The `dict(by_alias=True)` solution presented in #153 fails in this case, as it could be used to circumvent the Python keyword conflict, however:
1. it would require the `if_` alias value to be `if`, which also makes the model constructor harder to use, i.e. `Model(**{'if': value})`,
2. removes the ability to alias the `badly_named_field` to `better_name` in the constructor.

This means that a user has to fall back to overriding the `dict` method on the model to handle this case:
```python
class Model(pydantic.BaseModel):
    if_: str = pydantic.Field(None, alias='condition')
    badly_named_field: str = pydantic.Field(None, alias='better_name')

    def dict(self, *args, **kwargs):
        result = super().dict(*args, **kwargs)
        result['if'] = result.pop('if_')
        return result
```
This overloaded method may become more complicated if the user wants to conditionally add the `if` key to the resulting dict, i.e. , and if there are multiple properties that conflict with python keywords. "Complicated" is of course subjective, but given the elegance of the rest of the pydantic API, this seems like something that could be addressed.

# Proposal
Add a new `name` argument to `pydantic.Field` that aliases the model field name for serialisation, separate from the behaviour of `alias`. Essentially, a "this field should be called this but it can't be".

**Comment by samuelcolvin** [comment]
I think alias and (future) load and dump aliases are sufficient.

I think adding another concept of `name` is unnecessarily complicated.

**Reply by inkychris** [reply]
Oh, are load and dump aliases something in the works already? I was struggling to find issues relating to this topic so must have missed that.

**Reply by inkychris** [reply]
Ah yes, just found #624, that's basically exactly what I want!

**Reply by samuelcolvin** [reply]
Yes, thanks for finding the issue.

Sorry I'm in my mobile, so searching isn't that easy.
